Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseCollection.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    public abstract class BaseCollection : MonoBehaviour
+    {
+        #region public members
+        /// <summary>
+        /// Action called when collection is updated
+        /// </summary>
+        public Action<BaseCollection> OnCollectionUpdated;
+
+        /// <summary>
+        /// List of objects with generated data on the object.
+        /// </summary>
+        [HideInInspector]
+        public List<CollectionNode> NodeList = new List<CollectionNode>();
+
+        /// <summary>
+        /// Whether to treat inactive transforms as 'invisible'
+        /// </summary>
+        public bool IgnoreInactiveTransforms = true;
+
+        /// <summary>
+        /// Type of sorting to use.
+        /// </summary>
+        [Tooltip("Type of sorting to use")]
+        public SortTypeEnum SortType = SortTypeEnum.None;
+        #endregion
+
+
+        public virtual void UpdateCollection()
+        {
+
+            // Check for empty nodes and remove them
+            List<CollectionNode> emptyNodes = new List<CollectionNode>();
+
+            for (int i = 0; i < NodeList.Count; i++)
+            {
+                if (NodeList[i].transform == null || (IgnoreInactiveTransforms && !NodeList[i].transform.gameObject.activeSelf) || NodeList[i].transform.parent == null || !(NodeList[i].transform.parent.gameObject == this.gameObject))
```

Comment:
code style: I don't believe we have been using "this." in the toolkit.

Reply:
Old habits die hard. 😄